auslib/web/public/api.yml

```diff
@@ -0,0 +1,539 @@
+swagger: '2.0'
+info:
+  title: Balrog Public API
+  description: Balrog Application Update Service
+  version: "1.0.0"
+paths:
+  /update/1/{product}/{version}/{buildID}/{buildTarget}/{locale}/{channel}/update.xml:
+    get:
+      operationId: "auslib.web.public.client.get_update_blob_1"
+      description: Returns an update blob for the given parameters.
+      produces: ["text/xml"]
+      parameters:
+        - name: product
+          in: path
+          description: The name of the application requesting an update.
+          required: true
+          type: string
+        - name: version
+          in: path
+          description: The version of the application requesting an update.
+          required: true
+          type: string
+        - name: buildID
+          in: path
+          description: The build ID of the application requesting an update.
+          required: true
+          type: string
+        - name: buildTarget
+          in: path
+          description: The “build target” of the application requesting an update. This is usually related to the target platform the app was built for.
+          required: true
+          type: string
+        - name: locale
+          in: path
+          description: The locale of the application requesting an update.
+          required: true
+          type: string
+        - name: channel
+          in: path
+          description: The update channel of the application request an update.
+          required: true
+          type: string
+        # Underlying code depends on osVersion being set. Since this route only
+        # exists to support ancient queries, and all newer versions have osVersion
+        # in them it's easier to set this here than make the all of the underlying
+        # code support queries without it.
+        - name: queryVersion
+          in: path
+          description: queryVersion
+          required: true
+          type: integer
+          default: 2
```

Comment:
Yeah, it looks like the original implementation was buggy! I don't think it ends up getting used, but it's probably best to get it right.